src/main/java/com/telecom/antiphishing/controller/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,49 @@
+package com.telecom.antiphishing.controller;
+
+import com.telecom.antiphishing.exception.PhishingMessageException;
+import com.telecom.antiphishing.exception.UnexpectedResponseException;
+import com.telecom.antiphishing.model.ErrorMessage;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.time.LocalDateTime;
+
+/***
+ * @author - Kiryl Karpuk
+ */
+@ControllerAdvice
+public class ControllerExceptionHandler {
+
+    @ExceptionHandler(UnsupportedOperationException.class)
+    public ResponseEntity<ErrorMessage> catchWrongCommand(UnsupportedOperationException ex) {
+        ErrorMessage message =
+                new ErrorMessage(
+                        LocalDateTime.now(),
+                        ex.getMessage(),
+                        HttpStatus.BAD_REQUEST.getReasonPhrase());
+        return new ResponseEntity<>(message, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(UnexpectedResponseException.class)
+    public ResponseEntity<ErrorMessage> catchUnexpectedAnswer(UnexpectedResponseException ex) {
+        ErrorMessage message =
+                new ErrorMessage(
+                        LocalDateTime.now(),
+                        ex.getMessage(),
+                        HttpStatus.SERVICE_UNAVAILABLE.getReasonPhrase());
+        return new ResponseEntity<>(message, HttpStatus.SERVICE_UNAVAILABLE);
+    }
+
+    @ExceptionHandler(PhishingMessageException.class)
+    public ResponseEntity<ErrorMessage> catchPhishing(PhishingMessageException ex) {
+        ErrorMessage message =
+                new ErrorMessage(
+                        LocalDateTime.now(),
+                        ex.getMessage(),
+                        HttpStatus.UNAVAILABLE_FOR_LEGAL_REASONS.getReasonPhrase());
+        return new ResponseEntity<>(message, HttpStatus.UNAVAILABLE_FOR_LEGAL_REASONS);
```

Comment:
Super, że użyłeś i poszukałeś różnych kodów odpowiedzi - nie jestem do końca przekonany czy akurat ten kod jest adekwatny do sytuacji :thinking: Może mnie przekonasz? Dlaczego chcielibyśmy pokazać ten kod błędu osobie, która weryfikuje wiadomość za pomocą naszego API?

Reply:
Zgodnie z opisem tego status codu - A server operator has received a legal demand to deny access to a resource or to a set of resources that includes the requested resource. Wydawało mi się, że jest to najbardziej odpowiedni opis dla naszego zadania. Może gdyby blokowanie wiadomości phishingowych było wymagane przez prawo, użycie 451 miałoby więcej sensu.... Alternatywnie można użyć 403 forbidden, ale 403 dotyczy bardziej brakujących uprawnień niż nieudanej walidacji